### PR TITLE
[MTDSA-10273] - Retrieve a foreign property business annual summary models

### PR DESCRIPTION
**Pull request by DreamcastKnight**

### CODE REVIEW

File: app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignFhlEea/ForeignFhlEeaAllowances.scala [hunk 1]
```diff
@@ -16,14 +16,23 @@
...
+                                   propertyIncomeAllowance: Option[BigDecimal]
                                   )
 
 object ForeignFhlEeaAllowances {
-  implicit  val format: OFormat[ForeignFhlEeaAllowances] = Json.format[ForeignFhlEeaAllowances]
+  implicit  val writes: OWrites[ForeignFhlEeaAllowances] = Json.writes[ForeignFhlEeaAllowances]
+  implicit  val reads: Reads[ForeignFhlEeaAllowances] = (
+    (JsPath \ "annualInvestmentAllowance").readNullable[BigDecimal] and
+    (JsPath \ "otherCapitalAllowance").readNullable[BigDecimal] and
+    (JsPath \ "propertyAllowance").readNullable[BigDecimal] and
```
Comment (Thepworth95) [inline app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignFhlEea/ForeignFhlEeaAllowances.scala:34]: reading the propertyAllowance value twice?

File: app/v2/models/response/retrieveForeignPropertyAnnualSubmission/RetrieveForeignPropertyAnnualSubmissionResponse.scala [hunk 2]
```diff
@@ -17,16 +17,25 @@
...
+                                                            foreignFhlEea: Option[ForeignFhlEeaEntry],
+                                                            foreignNonFhlProperty: Option[Seq[ForeignPropertyEntry]])
 
 object RetrieveForeignPropertyAnnualSubmissionResponse extends HateoasLinks {
-  implicit  val format: OFormat[RetrieveForeignPropertyAnnualSubmissionResponse] = Json.format[RetrieveForeignPropertyAnnualSubmissionResponse]
+  implicit  val writes: OWrites[RetrieveForeignPropertyAnnualSubmissionResponse] = Json.writes[RetrieveForeignPropertyAnnualSubmissionResponse]
+  implicit  val reads: Reads[RetrieveForeignPropertyAnnualSubmissionResponse] = (
+    (JsPath \ "submittedOn").read[String] and
+      (JsPath \ "foreignFhlEea").readNullable[ForeignFhlEeaEntry] and
+      (JsPath \ "foreignProperty").readNullable[Seq[ForeignPropertyEntry]]
```
Comment (Thepworth95) [inline app/v2/models/response/retrieveForeignPropertyAnnualSubmission/RetrieveForeignPropertyAnnualSubmissionResponse.scala:37]: foreignNonFhlProperty downstream also returns `foreignNonFhlProperty` not `foreignProperty` so can be left as a standard read

File: app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignFhlEea/ForeignFhlEeaAllowances.scala [hunk 3]
```diff
@@ -16,14 +16,23 @@
 
 package v2.models.response.retrieveForeignPropertyAnnualSubmission.foreignFhlEea
 
-import play.api.libs.json.{Json, OFormat}
+import play.api.libs.json.{JsPath, Json, OWrites, Reads}
+import play.api.libs.functional.syntax._
 
 case class ForeignFhlEeaAllowances(annualInvestmentAllowance: Option[BigDecimal],
                                    otherCapitalAllowance: Option[BigDecimal],
```
Comment (Thepworth95) [inline app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignFhlEea/ForeignFhlEeaAllowances.scala:23]: fields dont match the spec. `propertyAllowance` should not be there and its `zeroEmissionsCarAllowance` should be

File: app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignProperty/ForeignPropertyAllowances.scala [hunk 4]
```diff
@@ -21,10 +21,11 @@ import play.api.libs.json.{Json, OFormat}
 case class ForeignPropertyAllowances(annualInvestmentAllowance: Option[BigDecimal],
                                      costOfReplacingDomesticItems: Option[BigDecimal],
                                      zeroEmissionsGoodsVehicleAllowance: Option[BigDecimal],
```
Comment (Thepworth95) [inline app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignProperty/ForeignPropertyAllowances.scala:23]: `zeroEmissionGoodsVehicleAllowance` not `zeroEmissionsGoodsVehicleAllowance` but is `zeroEmissionsGoodsVehicleAllowance` downstream so will need custom reads

File: app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignProperty/ForeignPropertyAllowances.scala [hunk 5]
```diff
@@ -21,10 +21,11 @@ import play.api.libs.json.{Json, OFormat}
...
-                                     propertyAllowance: Option[BigDecimal],
                                      otherCapitalAllowance: Option[BigDecimal],
-                                     structureAndBuildingAllowance: Option[BigDecimal],
-                                     electricChargePointAllowance:  Option[BigDecimal]
+                                     electricChargePointAllowance:  Option[BigDecimal],
+                                     zeroEmissionsCarAllowance: Option[BigDecimal],
+                                     propertyAllowance: Option[BigDecimal],
```
Comment (Thepworth95) [inline app/v2/models/response/retrieveForeignPropertyAnnualSubmission/foreignProperty/ForeignPropertyAllowances.scala:27]: `propertyIncomeAllowance` not `propertyAllowance` Maped from `propertyAllowance` downstream so will need custom reads

File: app/v2/models/response/retrieveForeignPropertyAnnualSubmission/RetrieveForeignPropertyAnnualSubmissionResponse.scala (hunk 2 above)
Comment (DreamcastKnight) [reply app/v2/models/response/retrieveForeignPropertyAnnualSubmission/RetrieveForeignPropertyAnnualSubmissionResponse.scala:37]: I checked the downstream spec and it has foreign property, not foreignNonFhlProperty

**Review by Thepworth95 — APPROVED** [review]

**Pull request merged**